### PR TITLE
Support length > 1 vectors in matchers

### LINKED ISSUE #50: select_helpers accept character vector?
Possible feature request though this may contradict #49 

Allow for natively passing a vector of strings to `starts_with`, `ends_with`, `contains` ... in order to avoid having to redefine the regex

Currently:
(1) Unwieldy manual list of single columns
```r
select(df,
       starts_with("a"),
       starts_with("b"),
       ends_with("c"),
       ends_with("d"),
       ...)
```

or

(2) Redefine matching regex for each `select_helper`
```r
long_list_of_select_vars <- c("a", "b", ...)
another_long_vector <- c("c", "d", ...)
select(df,
       matches(paste0("^(", paste(long_list_of_select_vars, collapse="|"), ")")),
       matches(paste0("(", paste(another_long_vector, collapse="|"), ")$")))
```

Request:
```r
select(df,
       starts_with(c("a", "b")),
       ends_with(c("c", "d")),
       ...)
```
which would have the same behaviour as (1)

[For context](https://stackoverflow.com/questions/42125049/using-starts-with-in-dplyr-with-a-vector-of-partial-column-names)

### PR DESCRIPTION
**Pull request by lionel-**
Closes #50.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,6 +1,14 @@
 
 # tidyselect (development)
 
+* `starts_with()`, `ends_with()`, `contains()`, and `matches()` now
```
Comment (hadley) [inline NEWS.md]: Maybe make it a little more clear that it’s always OR? (In the text, not just the code)

**Comment by hadley** [comment]
Should we consider this release 1.0.0?

**Comment by lionel-** [comment]
hmm, or should 1.0 be the release that features the new UI and adds support for predicate selections?

**Comment by hadley** [comment]
Should we just push on and do that for this release?

**Comment by lionel-** [comment]
Alright!

**Pull request merged**